tests/unit/build_providers/lxd/test_lxd.py

```diff
@@ -32,18 +31,24 @@
 if sys.platform == "linux":
     import pylxd
 
-
-class GetEnv(_base_provider.Provider):
-    def _get_env_command(self):
-        return ["env", "SNAPCRAFT_HAS_TTY=False"]
+LXD_RUN_COMMAND_PREFIX = [
+    "/snap/bin/lxc",
+    "exec",
+    "snapcraft-project-name",
+    "--",
+    "env",
+    "SNAPCRAFT_BUILD_ENVIRONMENT=managed-host",
+    "HOME=/root",
+    "SNAPCRAFT_HAS_TTY=False",
+]
 
 
 class EnvSetup(_base_provider.Provider):
     def _setup_environment(self):
         pass
 
 
-class LXDTestImpl(LXD, GetEnv, EnvSetup):
+class LXDTestImpl(LXD, EnvSetup):
```

Comment:
I would rather keep GetEnv and return nothing and have a simple test that makes sure that _get_env_command returns the right thing. Should keep things modular

Reply:
You're right.  I had this coupled with another change that sent me down that road before splitting this off, but the env flags should factored out. Will fix.

Reply:
Done